### PR TITLE
Test improvements

### PR DESCRIPTION
**Pull request by eliasnogueira**
# Test improvements

This PR adds test improvements in the `jnosql-communication-core` module.

## Added

- New Test Guideline

## Changed

- Added pitest library and a new profile to run the analysis
- Changed the following for all test classes at the `src/test/java`:
  - class structure
  - addition of the `@DisplayName` annotation
  - added data-driven methods in some classes
  - using assertions from the AssertJ library instead of JUnit 

## Removed

- Removed the `ValueTest` class and the test methods were already being verified by the `DefaultValueTest`

## Review before the merge

The following classes have `//TODO` comments and must be discussed before this PR can be merged.

### ByteValueReaderTest

The test `shouldConvert()` is using the `valueReader` method that, previously, was using the `Integer.class` to verify the possible conversions, but it was causing a `CastException`.
It was changed to `Byte.class` because it will return the converted value based on a byte value.

I believe the `Byte` is the correct class, but I need your guys input by

### LocalDateTimeReaderTest

When the assertions were changed by the soft ones I had to change the parameters, as assertions methods are comparing always the actual vs expected. Before we had expected vs actual.
Without the inversion, this test was throwing `NullPointerExceptions`

We need just a quick look at the `shouldConvert()` test to see if the actual objects make sense.

### OffsetDateTimeReaderTest

The test `shouldConvert()` was using the `OffSetDate` class to read, but I believe the correct one should be the `OffSetDateTime` class, which also makes sense in the context of this test.

Can you guys double-check and see if the `OffSetDateTime` class is the correct one to use?

### CODE REVIEW

**Comment by genie-jnosql** [comment]
Can one of the admins verify this patch?

File: README.adoc [hunk 1]
```diff
@@ -710,6 +710,10 @@ Here are the badges of this project:
 | image::https://sonarcloud.io/api/project_badges/measure?project=org.eclipse.jnosql%3Ajakarta-nosql-parent&metric=security_rating[window=_blank, link=https://sonarcloud.io/summary/new_code?id=org.eclipse.jnosql%3Ajakarta-nosql-parent]
 |===
 
+== Testing Guideline
+
+This project has a testing guideline that will help you understand Jakarta Data's testing practices.
+Please take a look at the _TESTING-GUIDELINE_ file.
```
Comment (otaviojava) [inline README.adoc:716]:
```suggestion
This project's testing guideline will help you understand Jakarta Data's testing practices.
Please take a look link:TESTING-GUIDELINE[at the file]
```

**Comment by otaviojava** [comment]
Amazing, there is a lot of improvement :)

**Comment by eliasnogueira** [comment]
@otaviojava what about the open questions in the PR description?

File: jnosql-communication/jnosql-communication-core/src/test/java/org/eclipse/jnosql/communication/reader/ByteValueReaderTest.java [hunk 2]
```diff
@@ -12,41 +12,49 @@
...
-        assertFalse(valueReader.test(Boolean.class));
+    @DisplayName("Should be incompatible")
+    void shouldValidateIncompatibility() {
+        assertSoftly(softly -> {
+            softly.assertThat(valueReader.test(AtomicBoolean.class)).as("AtomicBoolean is not compatible").isFalse();
+            softly.assertThat(valueReader.test(Boolean.class)).as("Boolean is not compatible").isFalse();
+        });
     }
 
+    //TODO: previously we had Integer, but it causes a CastException
```
Comment (otaviojava) [inline jnosql-communication/jnosql-communication-core/src/test/java/org/eclipse/jnosql/communication/reader/ByteValueReaderTest.java:48]: That is fine; we can keep it this way.
Indeed, that is the desirable result.

**Comment by otaviojava** [comment]
@eliasnogueira the PR is good enough, IMHO: let's merge it and open issues to solve on different PRs.

File: jnosql-communication/jnosql-communication-core/src/test/java/org/eclipse/jnosql/communication/reader/OffsetDateTimeReaderTest.java [hunk 3]
```diff
@@ -12,53 +12,62 @@
...
 import java.util.Date;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
 
-public class OffsetDateTimeReaderTest {
+class OffsetDateTimeReaderTest {
```
Comment (otaviojava) [inline jnosql-communication/jnosql-communication-core/src/test/java/org/eclipse/jnosql/communication/reader/OffsetDateTimeReaderTest.java:32]: @eliasnogueira, it makes sense.

**Comment by otaviojava** [comment]
@eliasnogueira could you solve the pom.xml conflict?

**Comment by eliasnogueira** [comment]
> @eliasnogueira could you solve the pom.xml conflict?

done :-)

File: pom.xml [hunk 4]
```diff
@@ -57,17 +57,15 @@
         </developer>
     </developers>
     <properties>
-        <apache.rat.version>0.12</apache.rat.version>
-        <jacoco.maven.version>0.8.9</jacoco.maven.version>
```
Comment (otaviojava) [inline pom.xml:61]:
```suggestion
        <jakarta.annotation.version>2.1.1</jakarta.annotation.version>
        <jacoco.maven.version>0.8.9</jacoco.maven.version>
```

@eliasnogueira you forget the jacoco properties.

**Comment by eliasnogueira** [comment]
@otaviojava I will go through the issues found during the build and adjust this PR

File: pom.xml [hunk 5]
```diff
@@ -76,16 +75,25 @@
...
-        <smallrye.config.version>3.1.3</smallrye.config.version>
+        <smallrye.config.version>3.2.1</smallrye.config.version>
+
+        <checkstyle.version>10.9.3</checkstyle.version>
+        <maven-gpg-plugin.version>1.6</maven-gpg-plugin.version>
+        <nexus-staging-maven-plugin.version>1.6.13</nexus-staging-maven-plugin.version>
+
+        <!-- testing libraries -->
+        <junit.version>5.9.2</junit.version>
+        <mockito.verson>5.2.0</mockito.verson>
```
Comment (otaviojava) [inline pom.xml:92]:
```suggestion
        <mockito.verson>4.11.0</mockito.verson>
```

There is a BK, we can fix on another PR.

**Pull request merged**